### DISCUSSION TITLE
How reference just single assembly from WinForms profile

### DISCUSSION THREAD

**Discussion by kant2002** [dotnet/winforms#5300] [Q&A] [answered]
I would like to reference just Accessibility.dll without hardcoding path to it in the project. How that's possible ?

**Comment by RussKie** [comment, marked as answer]
What are you trying to achieve?

Accessibility.dll is a part of Windows Desktop SDK, and it is implicitly referenced by the virtue of you declaring `<UseWindowsForms>true</UseWindowsForms>`.
It can be found under `C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\<version>\` and `C:\Program Files\dotnet\packs\Microsoft.WindowsDesktop.App.Ref\<version>\ref\net6.0\`.

**Reply by kant2002** [reply]
I trying to setup CI build for my library.
My library is somewhat special, since I create [fake projects](https://github.com/kant2002/WinFormsComInterop/blob/22cb08a799420fa652a5efa086e6871e7c17de95/WinFormsComInterop/WinFormsComInterop.csproj#L19-L27) for `System.Windows.Forms`, `System.Windows.Forms.Primitives` and others so using `<UseWindowsForms>true</UseWindowsForms>` probably too dangerous for me, or plainly would not work.

Right now I hardcode path to Accessibility.dll in `HintPath` which does not like when I run that ugly path in GitHub actions, since dotnet installed into different location. And I do not really want go route where I download dotnet in the some project location. That's too much infra for me. Any ideas? 

Also while I get your attention, how can I mark library on Nuget as library which is required WinForms?

**Reply by RussKie** [reply]
> how can I mark library on Nuget as library which is required WinForms?

An app doesn't directly reference Windows Forms or WPF libraries, instead the app gets those references through Windows Desktop SDK by setting UseWindowsForms or UseWPF. The easier way to see how it is done to compile an app with `/bl` switch, and then open the binlog in [MSBuild Binary and Structured Log Viewer](https://msbuildlog.com/).

I suppose if you library has a props/targets file that sets `<UseWindowsForms>true</UseWindowsForms>` it would force the consuming app to target Windows Forms.
That, and obviously target your library to `netX-windows`.

> My library is somewhat special, since I create [fake projects](https://github.com/kant2002/WinFormsComInterop/blob/22cb08a799420fa652a5efa086e6871e7c17de95/WinFormsComInterop/WinFormsComInterop.csproj#L19-L27) for `System.Windows.Forms`, `System.Windows.Forms.Primitives` and others

There are few thoughts here.
I'd reference Accessibility.dll from the output directory of System.Windows.Forms project (something like `$(OutDir)` variable, use MSBuild LogViewer to find the variable). Or, reference the assembly from the Windows Desktop ref folder (again  use MSBuild LogViewer to find the variable that contains the path).

**Reply by kant2002** [reply]
Thanks. I will try that.
